packages/common/http/test/response_spec.ts

```diff
@@ -94,4 +94,18 @@ describe('HttpResponse', () => {
       expect(clone.redirected).toBe(false);
     });
   });
+
+  describe('typings', () => {
+    it('should correctly narrow based on the type', () => {
+      const httpEvent: HttpEvent<any> = {
+        type: HttpEventType.DownloadProgress,
+        loaded: 100,
+        total: 200,
+      };
+
+      if (httpEvent.type === HttpEventType.DownloadProgress) {
+        httpEvent.partialText = 'Partial text';
```

Comment:
It's atypical to write to this field, so I'd propose swapping it around:

```suggestion
        const partialText: string = httpEvent.partialText;
```